packages/protocol/contracts/common/MetaTransactionWallet.sol

```diff
@@ -236,7 +365,101 @@ contract MetaTransactionWallet is
   }
 
   /**
-   * @notice Executes a transaction on behalf of the signer.`
+   * @notice Executes a refundable meta-transaction on behalf of the signer.
+   * @param destination The address to which the meta-transaction is to be sent.
+   * @param value The CELO value to be sent with the meta-transaction.
+   * @param data The data to be sent with the meta-transaction.
+   * @param maxGasPrice The maximum gas price the user is willing to pay.
+   * @param gasLimit The gas limit for entire relayed transaction.
+   * @param metaGasLimit The gas limit for just the meta-transaction.
```

Comment:
It is a bit unclear to me what the (user-signed) `metaGasLimit` is designed to protect against. My guess is that we need a way to assure relayers there is enough gas left-over to pay for the refund, and we need to the user to sign it because we wouldn't want the relayer to set it arbitrarily low and cause the user transaction to revert while still receiving a refund. Is that accurate? If so it would be helpful to clarify that.

Also, if that is true, and alternative mechanism to this would be to have a fixed constant amount of gas reserved for excuting the checks and refund logic. Was this considered?